### PR TITLE
feat: add experimental feature

### PR DESCRIPTION
**Pull request by ramniel**
contact-policy-4

***To make use of this feature by console it can be enabled like this:**
`localStorage.setItem('dopplerExperimental', JSON.stringify({ContactPolicy: true}));`

### CODE REVIEW

File: src/components/ContactPolicy/ContactPolicy.js [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
 import { FormattedMessageMarkdown } from '../../i18n/FormattedMessageMarkdown';
 import { Switch } from '../shared/Switch/Switch';
 import { FieldGroup, FieldItem, NumberField, SubmitButton } from '../form-helpers/form-helpers';
 import { Form, Formik } from 'formik';
 import useTimeout from '../../hooks/useTimeout';
+import { InjectAppServices } from '../../services/pure-di';
 
-export const ContactPolicy = () => {
+export const ContactPolicy = InjectAppServices(({ dependencies: { experimentalFeatures } }) => {
   const [active, toggleActive] = useState(false);
```
Comment (jcamposmk) [inline src/components/ContactPolicy/ContactPolicy.js:14]:
```suggestion
const [isActive, setIsActive] = useState(false);
```

Comment (jcamposmk) [reply src/components/ContactPolicy/ContactPolicy.js:14]: you can creae toggleActive and use setIsActive if you need

Comment (jcamposmk) [reply src/components/ContactPolicy/ContactPolicy.js:14]: Booleans examples: 

- isOpen
- canWrite
- hasFruit

File: src/components/ContactPolicy/ContactPolicy.js [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
   const [formSubmitted, setFormSubmitted] = useState(false);
```
Comment (jcamposmk) [inline src/components/ContactPolicy/ContactPolicy.js:15]:
```suggestion
const [isFormSubmitted, setIsFormSubmitted] = useState(false);
```

File: src/components/ContactPolicy/ContactPolicy.js [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
   const intl = useIntl();
   const _ = (id, values) => intl.formatMessage({ id: id }, values);
+  const isContactPolicyEnabled =
+    experimentalFeatures && experimentalFeatures.getFeature('ContactPolicy');
```
Comment (jcamposmk) [inline src/components/ContactPolicy/ContactPolicy.js:20]: you can move up `fieldNames` constant

File: src/components/ContactPolicy/ContactPolicy.js [hunk 4]
```diff
@@ -1,20 +1,22 @@
...
-export const ContactPolicy = () => {
-  const [active, toggleActive] = useState(false);
-  const [formSubmitted, setFormSubmitted] = useState(false);
+export const ContactPolicy = InjectAppServices(({ dependencies: { experimentalFeatures } }) => {
+  const [isActive, setIsActive] = useState(false);
+  const [isFormSubmitted, setIsFormSubmitted] = useState(false);
   const intl = useIntl();
   const _ = (id, values) => intl.formatMessage({ id: id }, values);
-
+  const isContactPolicyEnabled =
```
Comment (cbernat) [inline src/components/ContactPolicy/ContactPolicy.js:18]: When the api for features is ready we should store all features in session for example and then get this information from there.

File: src/components/ContactPolicy/ContactPolicy.js [hunk 5]
```diff
@@ -61,84 +63,97 @@ export const ContactPolicy = () => {
...
+            </div>
           </div>
-        </div>
+        ) : (
+          <div className="col-sm-12 col-md-8 col-lg-6 m-b-12">
+            <div className="dp-wrap-message dp-wrap-info">
+              <span className="dp-message-icon" />
+              <div className="dp-content-message">
+                <p>
+                  <FormattedMessage id="common.feature_no_available" />
```
Comment (cbernat) [inline src/components/ContactPolicy/ContactPolicy.js:150]: 💯

File: src/components/ContactPolicy/ContactPolicy.js [hunk 6]
```diff
@@ -1,20 +1,22 @@
...
 import { Form, Formik } from 'formik';
 import useTimeout from '../../hooks/useTimeout';
+import { InjectAppServices } from '../../services/pure-di';
 
-export const ContactPolicy = () => {
-  const [active, toggleActive] = useState(false);
-  const [formSubmitted, setFormSubmitted] = useState(false);
+export const ContactPolicy = InjectAppServices(({ dependencies: { experimentalFeatures } }) => {
+  const [isActive, setIsActive] = useState(false);
+  const [isFormSubmitted, setIsFormSubmitted] = useState(false);
```
Comment (cbernat) [inline src/components/ContactPolicy/ContactPolicy.js:15]: @jcamposmk @lucianaDilizio @aferraromedina opinions on this?

Comment (ramniel) [reply src/components/ContactPolicy/ContactPolicy.js:15]: https://github.com/FromDoppler/doppler-webapp/pull/1461#discussion_r663162841

File: src/components/ContactPolicy/ContactPolicy.js (hunk 4 above)
Comment (ramniel) [reply src/components/ContactPolicy/ContactPolicy.js:18]: Sure! This is just a temporary solution.

File: src/components/ContactPolicy/ContactPolicy.js [hunk 7]
```diff
@@ -1,20 +1,22 @@
...
 import { FieldGroup, FieldItem, NumberField, SubmitButton } from '../form-helpers/form-helpers';
 import { Form, Formik } from 'formik';
 import useTimeout from '../../hooks/useTimeout';
+import { InjectAppServices } from '../../services/pure-di';
 
-export const ContactPolicy = () => {
-  const [active, toggleActive] = useState(false);
-  const [formSubmitted, setFormSubmitted] = useState(false);
+export const ContactPolicy = InjectAppServices(({ dependencies: { experimentalFeatures } }) => {
+  const [isActive, setIsActive] = useState(false);
```
Comment (cbernat) [inline src/components/ContactPolicy/ContactPolicy.js:14]: How about this? We talked with the team and most of us are in favor of not using is.
```suggestion
  const [active, setActive] = useState(false);
```

File: src/components/ContactPolicy/ContactPolicy.js (hunk 6 above)
Comment (andresmoschini) [reply src/components/ContactPolicy/ContactPolicy.js:15]: Estuvimos discutiendo sobre este tema, porque @jcamposmk basandosé en _Clean Code_ propone usar siempre el prefijo `is` para los booleanos.

Pero en general, no es lo que estamos haciendo.

Así que por ahora preferimos que si está claro que es un estado (formas infinitivas o participio, `loading`, `closed`, `active`), no usar el `is`. En casos donde puede haber confusiones, sustantivos o verbos, ahí si, por ejemplo `isError` o `isOpen`.

Comment (cbernat) [reply src/components/ContactPolicy/ContactPolicy.js:15]: The same here
```suggestion
  const [formSubmitted, setFormSubmitted] = useState(false);
```

File: src/components/ContactPolicy/ContactPolicy.js (hunk 7 above)
Comment (ramniel) [reply src/components/ContactPolicy/ContactPolicy.js:14]: OK. Is this for all fields or just for 'active' field?

Comment (andresmoschini) [reply src/components/ContactPolicy/ContactPolicy.js:14]: Creo que este comentario puede responder tu pregunta: https://github.com/FromDoppler/doppler-webapp/pull/1461#discussion_r663217722

**Pull request merged**